rclcpp/include/rclcpp/parameter_value.hpp

```diff
@@ -166,13 +166,17 @@ class ParameterValue
 
   template<ParameterType type>
   constexpr
-  typename std::enable_if<type == ParameterType::PARAMETER_DOUBLE, const double &>::type
+  typename std::enable_if<type == ParameterType::PARAMETER_DOUBLE, double>::type
   get() const
   {
-    if (value_.type != rcl_interfaces::msg::ParameterType::PARAMETER_DOUBLE) {
-      throw ParameterTypeException(ParameterType::PARAMETER_DOUBLE, get_type());
+    // Allow integer values to be cast to double
```

Comment:
We should really document this somewhere, maybe in the high level `get/declare_parameter()` APIs, or maybe just right here.